### DISCUSSION TITLE
In unit test, how to read module event?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#380] [Questions] [answered]
### Discord user ID

0xaptos

### Describe your question in detail.

Take my simple message board contract as an example, I create a message in an object, I emit an event, is there any way to read the event in my unit test? I think current workaround is we create another function like `create_message_internal` that returns the message object, and call that in unit test. But I hate this approach because it feels so weird.

Ideally we can read the module event in unit test, so i can call the `get_message` in unit test passing message object to verify contract is correct.

```rust
    /// Create a new message
    public entry fun craete_message(sender: &signer, content: String) {
        let message_obj_constructor_ref = &object::create_object(@message_board_addr);
        let message_obj_signer = &object::generate_signer(message_obj_constructor_ref);
        let message = Message {
            creator: signer::address_of(sender),
            creation_timestamp: timestamp::now_seconds(),
            content,
        };
        move_to(message_obj_signer, message);

        event::emit(CreateMessageEvent {
            message_obj: object::object_from_constructor_ref(message_obj_constructor_ref),
            message,
        });
    }
```

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment, marked as answer]
You can use these test only functions to check emitted events.

https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/event.move#L21-L28
